### PR TITLE
task8

### PR DESCRIPTION
**Pull request by SafaaMagdyAwad**
Please don't hesitate to correct my mistakes.

### CODE REVIEW

File: app/Http/Controllers/ClassController.php [hunk 1]
```diff
@@ -4,12 +4,23 @@
 
 use App\Models\Classe;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/ClassController.php]: do you really need this Facade?

File: app/Http/Controllers/ClassController.php [hunk 2]
```diff
@@ -33,13 +44,19 @@ public function store(Request $request)
...
+            'image' => 'nullable|image|mimes:jpeg,png,jpg,gif,svg|max:2048',
             'capacity' => 'required|integer',
             'price' => 'required|decimal:0,3',
             'timeFrom' => 'required',
             'timeTo' => 'required|after:timeFrom',
         ]);
         // dd($request->all());
-       $validatedData['isFulled']=isset($request->isFulled);
+        $validatedData['isFulled']=isset($request->isFulled);
+        $image_name=(empty($request->image ))?"null":$this->upload_image($request->image);
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/ClassController.php]: my mentor recommends to me not to use  `empty` function, he claims that it has a bad internal implementation. use instead `isset()` or `$request->hasFile()`

File: app/Http/Controllers/ClassController.php [hunk 3]
```diff
@@ -70,18 +87,22 @@ public function edit(Classe $class)
...
             'capacity' => 'integer',
             'price' => 'decimal:0,3',
-            
             'timeTo' => '|after:timeFrom',
         ]);
        $validatedData['isFulled']=isset($request->isFulled);
        $validatedData['timeFrom']=$request->timeFrom;
-        // dd($validatedData);
+       // dd($validatedData);
+       $image_name=$request->hasFile('image')?$this->upload_image($request->image):$request->old_image;
```
Comment (Nora-Lootfy) [inline app/Http/Controllers/ClassController.php]: oh you made it the two ways😂

Comment (Nora-Lootfy) [reply app/Http/Controllers/ClassController.php]: there is another solution we will discuss in the session, but you did a pretty good work

File: database/migrations/2024_07_31_190904_add_to_classes_table.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<?php
```
Comment (Nora-Lootfy) [inline database/migrations/2024_07_31_190904_add_to_classes_table.php:1]: this way is pretty useful in production, when it is not possible to remove the data stored in the database,,

in development, use one migration file instead is better

File: resources/views/class.blade.php [hunk 5]
```diff
@@ -25,11 +25,20 @@
...
-                alt="" class="card-img"
+              <img src="{{ asset('assets/images/' . $class->image) }}"
```
Comment (Nora-Lootfy) [inline resources/views/class.blade.php:34]: nice😊

**Review by Nora-Lootfy — APPROVED** [review]
thanks for your hard working ya Safaa

File: app/Http/Controllers/ClassController.php (hunk 2 above)
Comment (SafaaMagdyAwad) [reply app/Http/Controllers/ClassController.php]: Thank you,  I'll  remember it. 🤍

File: app/Http/Controllers/ClassController.php (hunk 1 above)
Comment (SafaaMagdyAwad) [reply app/Http/Controllers/ClassController.php]: I was trying to remove the old image using it but i couldnt do it.

**Pull request closed without merging**